### PR TITLE
formal deprecation of old group class

### PR DESCRIPTION
**Pull request by fchapoton**
trying to finally get rid of the auld group class

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/groups/additive_abelian/additive_abelian_group.py [hunk 1]
```diff
@@ -6,10 +6,11 @@
 major differences are in the way elements are printed.
 """
 
-from sage.groups.old import AbelianGroup
+from sage.structure.parent import Parent
 from sage.modules.fg_pid.fgp_module import FGP_Module_class
 from sage.modules.fg_pid.fgp_element import FGP_Element
 from sage.rings.integer_ring import ZZ
+from sage.categories.commutative_additive_groups import CommutativeAdditiveGroups
```
Comment (mkoeppe) [inline src/sage/groups/additive_abelian/additive_abelian_group.py]: please keep sorted

File: src/sage/groups/additive_abelian/additive_abelian_group.py [hunk 2]
```diff
@@ -225,6 +226,7 @@ def __init__(self, cover, relations):
             True
         """
         FGP_Module_class.__init__(self, cover, relations)
+        Parent.__init__(self, category=CommutativeAdditiveGroups())
```
Comment (mkoeppe) [inline src/sage/groups/additive_abelian/additive_abelian_group.py]: Does this not initialize `Parent` twice? Should `FGP_Module_class` get a `category` argument?

File: src/sage/groups/group.pyx [hunk 3]
```diff
@@ -42,8 +42,7 @@ def is_Group(x):
         sage: is_Group("a string")
         False
     """
-    from sage.groups.old import Group as OldGroup
-    return isinstance(x, (Group, OldGroup))
+    return isinstance(x, Group)
```
Comment (mkoeppe) [inline src/sage/groups/group.pyx]: Is this correct?

Shouldn't we rather keep it and deprecate `is_Group` (see https://github.com/sagemath/sage/issues/32414)?

File: src/sage/groups/old.pyx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 """
-Base class for groups
+Deprecated Base class for groups
```
Comment (mkoeppe) [inline src/sage/groups/old.pyx]:
```suggestion
Deprecated base class for groups
```

File: src/sage/groups/old.pyx [hunk 5]
```diff
@@ -17,11 +17,11 @@ Base class for groups
 #                  https://www.gnu.org/licenses/
 # ****************************************************************************
 
-doc="""
-Base class for all groups
+doc = """
+Deprecated Base class for all groups
```
Comment (mkoeppe) [inline src/sage/groups/old.pyx]:
```suggestion
Deprecated base class for all groups
```

File: src/sage/groups/group.pyx (hunk 3 above)
Comment (fchapoton) [reply src/sage/groups/group.pyx]: I will deprecate the version with only `Group`.

File: src/sage/categories/action.pyx [hunk 6]
```diff
@@ -381,10 +381,10 @@ cdef class InverseAction(Action):
     def __init__(self, Action action):
         G = action.G
         try:
-            from sage.groups.group import is_Group
+            from sage.groups.group import Group
             # We must be in the case that parent(~a) == parent(a)
             # so we can invert in _call_ code below.
-            if (is_Group(G) and G.is_multiplicative()) or G.is_field():
+            if (isinstance(G, Group) and G.is_multiplicative()) or G.is_field():
```
Comment (mkoeppe) [inline src/sage/categories/action.pyx:387]: Is not testing `OldGroup` here not a breaking change?

Comment (fchapoton) [reply src/sage/categories/action.pyx:387]: I consider that `OldGroup` was implicitly deprecated by its very name.

Comment (mkoeppe) [reply src/sage/categories/action.pyx:387]: OK, I agree.

**Pull request merged**